src/Symfony/Component/RateLimiter/SlidingWindow.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ *
+ * @experimental in 5.2
+ */
+final class SlidingWindow implements LimiterStateInterface
+{
+    private $id;
+    private $hitCount = 0;
+    private $hitCountForLastWindow;
+    private $intervalInSeconds;
+    private $createdAt;
+    private $cached = true;
+
+    public function __construct(string $id, int $intervalInSeconds, int $lastWindowHitCount = 0)
+    {
+        $this->id = $id;
+        $this->intervalInSeconds = $intervalInSeconds;
+        $this->hitCountForLastWindow = $lastWindowHitCount;
+        $this->createdAt = time();
+        $this->cached = false;
+    }
+
+    public function getId(): string
+    {
+        return $this->id;
+    }
+
+    /**
+     * Store for the rest of this time frame and next.
+     */
+    public function getExpirationTime(): ?int
+    {
+        if ($this->cached) {
+            return null;
+        }
```

Comment:
Isn't `$cached` always false? Given that we serialize the whole object and set it to `false` in the constructor?

Looking at the code, I would also expect to just always return `2 * $this->intervalInSeconds` here.

Reply:
You are correct. 

I'll add back the sleep method